instrumentation/opentelemetry-instrumentation-botocore/src/opentelemetry/instrumentation/botocore/__init__.py

```diff
@@ -67,7 +67,7 @@
 def _patched_endpoint_prepare_request(wrapped, instance, args, kwargs):
     request = args[0]
     headers = request.headers
-    propagators.inject(type(headers).__setitem__, headers)
+    propagators.inject(type(headers).add_header, headers)
```

Comment:
I don't have a lot of context on this, but we should probably add at least a test for the case you're running into.

From https://github.com/open-telemetry/opentelemetry-python-contrib/issues/262, it sounds like the most relevant part is the DyanamoDB API call. have you tried adding a mock DynamiDB api call (moto [looks to support dynamodb](https://github.com/spulec/moto/tree/master/moto/dynamodb))? It seems like a call that maps what you're executing in your service should trigger the error.

Reply:
Taking a closer look, the part I'm a little unclear on is, as you mentioned, what is actually causing this. I have a feeling something was injected that should not have been by the propagator. I'll comment more on the issue.